core/src/main/java/io/grpc/internal/DelayedStream.java

```diff
@@ -306,14 +306,11 @@ public void cancel(final Status reason) {
     checkState(listener != null, "May only be called after start");
```

Comment:
Separate note: there might be a race at delayedClient `shutdownNow()` to make `start()` call after `cancel()`, which may cause shutdownNow fail?

Reply:
Yes, `shutdownNow()` could call `cancel()` before `start()` which is now disallowed. Fixing that is not trivial; I'll need to look into a few options.

Reply:
For `shutdownNow()`, we can do `runnable = pendingStream.setStream(new FailingStream(status))` before `pendingStream.cancel(status)`

Reply:
If we keep reusing `cancel()` for force shutdown then we can't avoid competing with customer call and there is no way we can guarantee the call order, and we would have to support that. I feel it would be fair to have `delayedStream` take the responsibility instead of `shutdownNow()`. Or we use another method to do force shutdown to not mix with `cancel()`, but that would require same about of change. This problem is originally there so I guess we should fix that separately.

Reply:
I was wrong. Fixing it was trivial! "Use setStream() instead of cancel()." Thank you @YifeiZhuang for noticing.